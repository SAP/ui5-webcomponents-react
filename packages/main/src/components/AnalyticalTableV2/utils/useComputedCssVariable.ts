import type { RefObject } from 'react';
import { useEffect, useState } from 'react';

export function useComputedCssVariable<T extends HTMLElement>(ref: RefObject<T>, variableName: string): number | null {
  const [value, setValue] = useState<number | null>(null);

  useEffect(() => {
    if (ref.current) {
      const computedStyles = getComputedStyle(ref.current);
      const variableValue = computedStyles.getPropertyValue(variableName).trim();

      // Attempt to parse the value as a number
      const parsedValue = parseFloat(variableValue);
      setValue(isNaN(parsedValue) ? null : parsedValue);
    }
  }, [variableName, ref]);

  return value;
}
