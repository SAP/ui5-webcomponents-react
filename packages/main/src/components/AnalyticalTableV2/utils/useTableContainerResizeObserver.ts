import type { RefObject } from 'react';
import { useState, useEffect } from 'react';

export const useTableContainerResizeObserver = (tableContainerRef: RefObject<HTMLDivElement>) => {
  const [tableWidth, setTableWidth] = useState(0);
  const [horizontalScrollbarHeight, setHorizontalScrollbarHeight] = useState(0);

  useEffect(() => {
    const tableContainer = tableContainerRef.current;

    const resizeObserver = new ResizeObserver((entries) => {
      for (const entry of entries) {
        if (entry.borderBoxSize) {
          const { borderBoxSize, contentBoxSize } = entry;
          const borderBoxHeight = borderBoxSize[0].blockSize;
          const contentBoxHeight = contentBoxSize[0].blockSize;
          if (borderBoxHeight > contentBoxHeight) {
            setHorizontalScrollbarHeight(borderBoxHeight - contentBoxHeight);
          } else {
            setHorizontalScrollbarHeight(0);
          }

          const borderBoxWidth = borderBoxSize[0].inlineSize;
          setTableWidth(borderBoxWidth);
        }
      }
    });

    if (tableContainer) {
      resizeObserver.observe(tableContainer, { box: 'border-box' });
    }

    return () => {
      if (tableContainer) {
        resizeObserver.unobserve(tableContainer);
      }
    };
  }, [tableContainerRef]);

  return { tableWidth, horizontalScrollbarHeight };
};
