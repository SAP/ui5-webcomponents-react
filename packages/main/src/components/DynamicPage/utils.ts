export const DynamicPageCssVariables = {
  headerDisplay: '--_ui5wcr_DynamicPage_header_display',
  titleFontSize: '--_ui5wcr_DynamicPage_title_fontsize'
};
