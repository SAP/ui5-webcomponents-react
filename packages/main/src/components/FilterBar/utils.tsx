import { cloneElement } from 'react';

const inputTagNames = new Set([
  'UI5-COMBOBOX',
  'UI5-DATE-PICKER',
  'UI5-DATERANGE-PICKER',
  'UI5-DATETIME-PICKER',
  'UI5-DURATION-PICKER',
  'UI5-INPUT',
  'UI5-MULTI-INPUT',
  'UI5-RATING-INDICATOR',
  'UI5-STEP-INPUT',
  'UI5-TIME-PICKER'
]);

export const filterValue = (ref, child) => {
  const tagName = ref.tagName;
  let filterItemProps = {};
  if (inputTagNames.has(tagName)) {
    filterItemProps = { value: ref.value };
  }
  if (tagName === 'UI5-SELECT' || tagName === 'UI5-MULTI-COMBOBOX') {
    const selectedIndices = Array.from(ref.children as HTMLCollectionOf<any>)
      .map((item, index) => (item.selected ? index : false))
      .filter((el) => el !== false);
    const selectedIndicesSet = new Set(selectedIndices);
    const options = child.props.children.props.children?.map((item, index) => {
      if (selectedIndicesSet.has(index)) {
        return cloneElement(item, { selected: true });
      }
      return cloneElement(item, { selected: false });
    });
    filterItemProps = { children: options };
  }
  if (tagName === 'UI5-SWITCH' || tagName === 'UI5-CHECKBOX') {
    filterItemProps = { checked: ref.checked };
  }
  return filterItemProps;
};

export const renderSearchWithValue = (renderSearchElement, searchValue) => {
  return cloneElement(renderSearchElement, { value: searchValue ?? renderSearchElement?.props?.value });
};
