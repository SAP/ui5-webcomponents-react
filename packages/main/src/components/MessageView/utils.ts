import iconAlert from '@ui5/webcomponents-icons/dist/alert.js';
import iconError from '@ui5/webcomponents-icons/dist/error.js';
import iconInformation from '@ui5/webcomponents-icons/dist/information.js';
import iconSysEnter from '@ui5/webcomponents-icons/dist/sys-enter-2.js';
import { ValueState } from '../../enums/ValueState';

export const getIconNameForType = (type: ValueState | keyof typeof ValueState): string => {
  switch (type) {
    case ValueState.Error:
    case 'Error':
      return iconError;
    case ValueState.Success:
    case 'Success':
      return iconSysEnter;
    case ValueState.Warning:
    case 'Warning':
      return iconAlert;
    default:
      return iconInformation;
  }
};
