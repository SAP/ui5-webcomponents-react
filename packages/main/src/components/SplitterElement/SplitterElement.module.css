.splitterElement {
  display: flex;
  overflow: hidden;
  position: relative;
  will-change: flex-basis;
  min-width: 0px;
  min-height: 0px;
}
