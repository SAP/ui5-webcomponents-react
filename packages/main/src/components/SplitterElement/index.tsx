import { Device, useIsomorphicLayoutEffect, useSyncRef } from '@ui5/webcomponents-react-base';
import clsx from 'clsx';
import React, { CSSProperties, forwardRef, ReactNode, useContext, useEffect, useState } from 'react';
import { createUseStyles } from 'react-jss';
import { CommonProps } from '../../interfaces/CommonProps';
import { SplitterLayoutContext } from '../../internal/SplitterLayoutContext';

const useStyles = createUseStyles(
  {
    splitterElement: {
      display: 'flex',
      overflow: 'hidden',
      position: 'relative',
      willChange: 'flex-basis',
      minWidth: '0px',
      minHeight: '0px'
    }
  },
  { name: 'SplitterElement' }
);

export interface SplitterElementPropTypes extends CommonProps {
  /**
   * Defines whether a resizer element is displayed.
   *
   * If the next `SplitterElement` has the prop `resizable={false}`, no resizer element will be shown after this
   * `SplitterElement`. The resizer element is only shown when all siblings of the resizer are resizable.
   * Default value of resizable is `true`.
   */
  resizable?: boolean;
  /**
   * Defines the initial size of the `SplitterElement`.
   */
  size?: CSSProperties['width'] | CSSProperties['height'];
  /**
   * Defines the minimum size of the `SplitterElement`. The resizer element stops when the minimum size is reached.
   */
  minSize?: number;
  /**
   * Defines the content which is shown inside the `SplitterElement`.
   */
  children?: ReactNode | ReactNode[];
}

/**
 * The `SplitterElement` holds the component of the content area. Allowed size values are of the type css property width or
 * height according to the orientation of the `SplitterLayout`. If `size` isn't passed to the element, the width or
 * height of the content area will be calculated automatically according to the size of the given `SplitterLayout`.
 * The `minSize` defines the minimum width or height of the area and is set to 0 when no minimum size is given, so the
 * content can be completely collapsed.
 */
const SplitterElement = forwardRef<HTMLDivElement, SplitterElementPropTypes>((props, ref) => {
  const { children, style, className, minSize, size, resizable: _0, ...rest } = props;
  const [componentRef, splitterElementRef] = useSyncRef(ref);
  const { vertical, reset } = useContext(SplitterLayoutContext);
  const safariStyles = Device.isSafari() ? { width: 'min-content', flex: '1 0 auto' } : {};
  const defaultFlexStyles = size !== 'auto' ? { flex: `0 0 ${size}` } : { flex: '1 0 min-content', ...safariStyles };
  const [flexStyles, setFlexStyles] = useState(defaultFlexStyles);
  const [flexBasisApplied, setFlexBasisApplied] = useState(false);
  const classes = useStyles();

  useEffect(() => {
    const elementObserver = new ResizeObserver(([element]) => {
      if (element.target.getBoundingClientRect().width !== 0 && !flexBasisApplied) {
        const resetSafariStyles = Device.isSafari() ? { width: 'unset' } : {};
        setFlexStyles({ flex: `0 0 ${element.target.getBoundingClientRect().width}px`, ...resetSafariStyles });
        setFlexBasisApplied(true);
      }
    });
    if (size === 'auto' && splitterElementRef.current) {
      elementObserver.observe(splitterElementRef.current);
    } else {
      setFlexStyles({ flex: `0 0 ${size}` });
    }

    return () => {
      elementObserver.disconnect();
    };
  }, [size, flexBasisApplied]);

  useIsomorphicLayoutEffect(() => {
    if (reset) {
      setFlexStyles(undefined);
      setFlexBasisApplied(false);
    }
  }, [reset, size]);

  useIsomorphicLayoutEffect(() => {
    if (flexStyles === undefined) {
      setFlexStyles(defaultFlexStyles);
    }
  }, [flexStyles]);

  return (
    <div
      ref={componentRef}
      className={clsx(classes.splitterElement, classes[vertical ? 'vertical' : 'horizontal'], className)}
      style={{
        minHeight: vertical && minSize ? `${minSize}px` : undefined,
        minWidth: !vertical && minSize ? `${minSize}px` : undefined,
        ...flexStyles,
        ...style
      }}
      {...rest}
      data-min-size={minSize}
    >
      {children}
    </div>
  );
});

SplitterElement.defaultProps = {
  minSize: 0,
  size: 'auto'
};

SplitterElement.displayName = 'SplitterElement';

export { SplitterElement };
