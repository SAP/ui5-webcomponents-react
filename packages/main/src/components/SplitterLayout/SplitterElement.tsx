import { ThemingParameters } from '@ui5/webcomponents-react-base/dist/ThemingParameters';
import { FlexBoxJustifyContent } from '@ui5/webcomponents-react/dist/FlexBoxJustifyContent';
import { forwardRef, ReactNode, ReactNodeArray, Ref } from 'react';
import { ReflexElement } from 'react-reflex';

export interface SplitterElementPropTypes {
  resizable?: boolean;
  size?: number;
  minSize?: number;
  maxSize?: number;
  justifyContent?: FlexBoxJustifyContent | keyof typeof FlexBoxJustifyContent;
  children?: ReactNode | ReactNodeArray;
}

const SplitterElement = forwardRef((props: SplitterElementPropTypes, ref: Ref<HTMLDivElement>) => {
  const { size, minSize, maxSize, children, justifyContent } = props;

  return (
    <ReflexElement
      ref={ref}
      size={size}
      minSize={minSize}
      maxSize={maxSize}
      style={{
        display: 'flex',
        justifyContent,
        backgroundColor: ThemingParameters.sapBackgroundColor
      }}
      {...props}
    >
      {children}
    </ReflexElement>
  );
});

SplitterElement.displayName = 'SplitterElement';

export { SplitterElement };
