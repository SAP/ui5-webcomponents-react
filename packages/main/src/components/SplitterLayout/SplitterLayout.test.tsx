import { fireEvent, render, renderRtl, screen } from '@shared/tests';
import { createCustomPropsTest } from '@shared/tests/utils';
import { ThemingParameters } from '@ui5/webcomponents-react-base';
import React from 'react';
import { Button } from '../../webComponents/Button';
import { SplitterElement } from '../SplitterElement';
import { SplitterLayout } from './index';

const keyDownOptions = (direction) => ({ code: `Arrow${direction}`, key: `Arrow${direction}` });

describe('SplitterLayout', () => {
  beforeEach(() => {
    // @ts-ignore
    window.HTMLElement.prototype.getBoundingClientRect = () => {
      return {
        width: 200,
        height: 800,
        top: 0,
        left: 0,
        bottom: 0,
        right: 0,
        x: 0,
        y: 0
      };
    };
  });

  test('Render Vertical SplitterLayout with multiple SplitterElements', () => {
    const onClick = jest.fn();
    const { asFragment } = render(
      <SplitterLayout vertical={true} style={{ width: '800px', height: '400px' }} data-testid={'SplitterLayout'}>
        <SplitterElement size={'200px'} data-testid={'SplitterElement1'}>
          <Button data-testid={'Button'} onClick={onClick}>
            Button 1
          </Button>
        </SplitterElement>
        <SplitterElement minSize={300} size={'400px'} data-testid={'SplitterElement2'}>
          <Button>Button 2</Button>
        </SplitterElement>
        <SplitterElement resizable={false} data-testid={'SplitterElement3'}>
          <Button>Button 3</Button>
        </SplitterElement>
      </SplitterLayout>
    );
    const AllSplitter = document.querySelectorAll('[role="separator"]');

    fireEvent.click(screen.getByTestId('Button'));
    fireEvent.click(AllSplitter[0]);

    expect(AllSplitter.length).toBe(1);
    expect(AllSplitter[0]).toHaveStyle(`border: 1px dotted ${ThemingParameters.sapHighlightColor}`);

    expect(screen.getByTestId('SplitterLayout')).toHaveStyle('width: 800px; height: 400px');
    expect(screen.getByTestId('SplitterElement1')).toHaveStyle('flex: 0 0 auto');
    expect(screen.getByTestId('SplitterElement2')).toHaveStyle('flex: 0 0 auto; min-height: 300px');
    expect(screen.getByTestId('SplitterElement2').dataset.minSize).toBe('300');
    expect(screen.getByTestId('SplitterElement3')).toHaveStyle('flex: 1 0 auto');
    expect(onClick).toHaveBeenCalled();

    expect(asFragment()).toMatchSnapshot();
  });

  test('Render Vertical SplitterLayout with only one SplitterElements', () => {
    const onClick = jest.fn();
    const { asFragment } = render(
      <SplitterLayout vertical={true} style={{ width: '700px', height: '400px' }} data-testid={'SplitterLayout'}>
        <SplitterElement size={'200px'} data-testid={'SplitterElement'}>
          <Button data-testid={'Button'} onClick={onClick}>
            Button 1
          </Button>
        </SplitterElement>
      </SplitterLayout>
    );
    const AllSplitter = document.querySelectorAll('[role="separator"]');

    fireEvent.click(screen.getByTestId('Button'));

    expect(AllSplitter.length).toBe(0);
    expect(screen.getByTestId('SplitterLayout')).toHaveStyle('width: 700px; height: 400px');
    expect(screen.getByTestId('SplitterLayout').dataset.splitterVertical).toBe('true');
    expect(screen.getByTestId('SplitterElement')).toHaveStyle('flex: 0 0 auto');
    expect(onClick).toHaveBeenCalled();

    expect(asFragment()).toMatchSnapshot();
  });

  test('Render Horizontal SplitterLayout with multiple SplitterElements', () => {
    const { asFragment } = render(
      <SplitterLayout data-testid={'Layout'} style={{ width: '80vw', height: '600px' }}>
        <SplitterElement>
          <Button>Button 1</Button>
        </SplitterElement>
        <SplitterElement data-testid={'Element'} minSize={50} size={'100px'}>
          <Button>Button 2</Button>
        </SplitterElement>
        <SplitterElement>
          <Button>Button 3</Button>
        </SplitterElement>
      </SplitterLayout>
    );

    const AllSplitter = document.querySelectorAll('[role="separator"]');

    expect(AllSplitter.length).toBe(2);
    expect(screen.getByTestId('Layout')).toHaveStyle('width: 80vw; height: 600px');
    expect(screen.getByTestId('Layout').dataset.splitterVertical).toBe(undefined);
    expect(screen.getByTestId('Element')).toHaveStyle('flex: 0 0 auto; min-width: 50px');
    expect(screen.getByTestId('Element').dataset.minSize).toBe('50');
    expect(asFragment()).toMatchSnapshot();
  });

  test('Render SplitterLayout without props and children', () => {
    const { asFragment } = render(<SplitterLayout></SplitterLayout>);

    expect(asFragment()).toMatchSnapshot();
  });

  test('Splitter click, move, focus - vertical', () => {
    const { getByTestId } = render(
      <SplitterLayout vertical style={{ width: '800px', height: '800px' }} data-testid={'SplitterLayout'}>
        <SplitterElement data-testid={'SplitterElement1'}>Content 1</SplitterElement>
        <SplitterElement data-testid={'SplitterElement2'}>Content 2</SplitterElement>
        <SplitterElement data-testid={'SplitterElement3'} size={'200px'}>
          Content 3
        </SplitterElement>
        <SplitterElement data-testid={'SplitterElement4'} size={'200px'}>
          Content 4
        </SplitterElement>
      </SplitterLayout>
    );
    const AllSplitter = document.querySelectorAll('[role="separator"]');

    // click focuses the corresponding splitter
    fireEvent.click(AllSplitter[2]);
    expect(document.activeElement).toBe(AllSplitter[2]);

    fireEvent.keyDown(document.activeElement, keyDownOptions('Left'));

    expect(getByTestId('SplitterElement3').style.flexBasis).toBe('780px');
    expect(getByTestId('SplitterElement4').style.flexBasis).toBe('820px');

    fireEvent.keyDown(document.activeElement, keyDownOptions('Right'));
    fireEvent.keyDown(document.activeElement, keyDownOptions('Right'));

    expect(getByTestId('SplitterElement3').style.flexBasis).toBe('820px');
    expect(getByTestId('SplitterElement4').style.flexBasis).toBe('780px');
  });

  test('Splitter click, move, focus - horizontal', () => {
    const { getByTestId } = render(
      <SplitterLayout style={{ width: '800px', height: '800px' }} data-testid={'SplitterLayout'}>
        <SplitterElement data-testid={'SplitterElement1'}>Content 1</SplitterElement>
        <SplitterElement data-testid={'SplitterElement2'}>Content 2</SplitterElement>
        <SplitterElement data-testid={'SplitterElement3'} size={'800px'}>
          Content 3
        </SplitterElement>
        <SplitterElement data-testid={'SplitterElement4'} size={'800px'}>
          Content 4
        </SplitterElement>
      </SplitterLayout>
    );
    const AllSplitter = document.querySelectorAll('[role="separator"]');

    // click focuses the corresponding splitter
    fireEvent.click(AllSplitter[2]);
    expect(document.activeElement).toBe(AllSplitter[2]);

    fireEvent.keyDown(document.activeElement, keyDownOptions('Up'));

    expect(getByTestId('SplitterElement3').style.flex).toBe('0 0 180px');
    expect(getByTestId('SplitterElement4').style.flex).toBe('0 0 220px');

    fireEvent.keyDown(document.activeElement, keyDownOptions('Down'));
    fireEvent.keyDown(document.activeElement, keyDownOptions('Down'));

    expect(getByTestId('SplitterElement3').style.flex).toBe('0 0 220px');
    expect(getByTestId('SplitterElement4').style.flex).toBe('0 0 180px');
  });

  test('Splitter click, move, focus - vertical - RTL', () => {
    const { getByTestId } = renderRtl(
      <SplitterLayout style={{ width: '800px', height: '800px' }} data-testid={'SplitterLayout'}>
        <SplitterElement data-testid={'SplitterElement1'}>Content 1</SplitterElement>
        <SplitterElement data-testid={'SplitterElement2'}>Content 2</SplitterElement>
        <SplitterElement data-testid={'SplitterElement3'} size={'200px'}>
          Content 3
        </SplitterElement>
        <SplitterElement data-testid={'SplitterElement4'} size={'200px'}>
          Content 4
        </SplitterElement>
      </SplitterLayout>
    );
    const AllSplitter = document.querySelectorAll('[role="separator"]');

    // click focuses the corresponding splitter
    fireEvent.click(AllSplitter[2]);
    expect(document.activeElement).toBe(AllSplitter[2]);

    fireEvent.keyDown(document.activeElement, keyDownOptions('Left'));

    expect(getByTestId('SplitterElement3').style.flexBasis).toBe('220px');
    expect(getByTestId('SplitterElement4').style.flexBasis).toBe('180px');

    fireEvent.keyDown(document.activeElement, keyDownOptions('Right'));
    fireEvent.keyDown(document.activeElement, keyDownOptions('Right'));

    expect(getByTestId('SplitterElement3').style.flexBasis).toBe('180px');
    expect(getByTestId('SplitterElement4').style.flexBasis).toBe('220px');
  });

  createCustomPropsTest(SplitterLayout);
});
