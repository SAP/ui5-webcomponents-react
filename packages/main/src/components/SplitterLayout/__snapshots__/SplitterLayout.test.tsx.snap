// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitterLayout Render Horizontal SplitterLayout with multiple SplitterElements 1`] = `
<DocumentFragment>
  <div
    class="SplitterLayout-splitterLayout"
    data-testid="Layout"
    style="flex-direction: row; width: 80vw; height: 600px;"
  >
    <div
      class="SplitterElement-splitterElement"
      data-min-size="0"
    >
      <ui5-button
        design="Default"
        ui5-button=""
      >
        Button 1
      </ui5-button>
    </div>
    <div
      aria-label="Press ARROW keys to move"
      aria-orientation="horizontal"
      class="Splitter-splitter"
      data-splitter-vertical="horizontal"
      role="separator"
      tabindex="0"
      title="Press ARROW keys to move"
    >
      <div
        class="Splitter-lineBefore"
      />
      <ui5-icon
        class="Splitter-icon"
        design="Default"
        name="vertical-grip"
        ui5-icon=""
      />
      <div
        class="Splitter-lineAfter"
      />
    </div>
    <div
      class="SplitterElement-splitterElement"
      data-min-size="50"
      data-testid="Element"
      style="min-width: 50px;"
    >
      <ui5-button
        design="Default"
        ui5-button=""
      >
        Button 2
      </ui5-button>
    </div>
    <div
      aria-label="Press ARROW keys to move"
      aria-orientation="horizontal"
      class="Splitter-splitter"
      data-splitter-vertical="horizontal"
      role="separator"
      tabindex="0"
      title="Press ARROW keys to move"
    >
      <div
        class="Splitter-lineBefore"
      />
      <ui5-icon
        class="Splitter-icon"
        design="Default"
        name="vertical-grip"
        ui5-icon=""
      />
      <div
        class="Splitter-lineAfter"
      />
    </div>
    <div
      class="SplitterElement-splitterElement"
      data-min-size="0"
    >
      <ui5-button
        design="Default"
        ui5-button=""
      >
        Button 3
      </ui5-button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SplitterLayout Render SplitterLayout without props and children 1`] = `
<DocumentFragment>
  <div
    class="SplitterLayout-splitterLayout"
    style="flex-direction: row;"
  />
</DocumentFragment>
`;

exports[`SplitterLayout Render Vertical SplitterLayout with multiple SplitterElements 1`] = `
<DocumentFragment>
  <div
    class="SplitterLayout-splitterLayout"
    data-splitter-vertical="true"
    data-testid="SplitterLayout"
    style="flex-direction: column; width: 800px; height: 400px;"
  >
    <div
      class="SplitterElement-splitterElement"
      data-min-size="0"
      data-testid="SplitterElement1"
    >
      <ui5-button
        data-testid="Button"
        design="Default"
        ui5-button=""
      >
        Button 1
      </ui5-button>
    </div>
    <div
      aria-label="Press ARROW keys to move"
      aria-orientation="vertical"
      class="Splitter-splitter"
      data-splitter-vertical="vertical"
      role="separator"
      tabindex="0"
      title="Press ARROW keys to move"
    >
      <div
        class="Splitter-lineBefore"
      />
      <ui5-icon
        class="Splitter-icon"
        design="Default"
        name="horizontal-grip"
        ui5-icon=""
      />
      <div
        class="Splitter-lineAfter"
      />
    </div>
    <div
      class="SplitterElement-splitterElement"
      data-min-size="300"
      data-testid="SplitterElement2"
      style="min-height: 300px; flex: 0 0 400px;"
    >
      <ui5-button
        design="Default"
        ui5-button=""
      >
        Button 2
      </ui5-button>
    </div>
    <div
      class="SplitterElement-splitterElement"
      data-min-size="0"
      data-testid="SplitterElement3"
    >
      <ui5-button
        design="Default"
        ui5-button=""
      >
        Button 3
      </ui5-button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SplitterLayout Render Vertical SplitterLayout with only one SplitterElement 1`] = `
<DocumentFragment>
  <div
    class="SplitterLayout-splitterLayout"
    data-splitter-vertical="true"
    data-testid="SplitterLayout"
    style="flex-direction: column; width: 700px; height: 400px;"
  >
    <div
      class="SplitterElement-splitterElement"
      data-min-size="0"
      data-testid="SplitterElement"
      style="flex: 0 0 200px;"
    >
      <ui5-button
        data-testid="Button"
        design="Default"
        ui5-button=""
      >
        Button 1
      </ui5-button>
    </div>
  </div>
</DocumentFragment>
`;
